url8d3c4d17f5_catseye_JaC64_tgz-pJ8-JaC64J8/src/C64Applet.java

```diff
@@ -0,0 +1,665 @@
+  // Utils
+  // -------------------------------------------------------------------
+
+
+  private int getParameterAsInt(String paramName, int defVal) {
+    String val = getParameter(paramName);
+    System.out.println(paramName + " = " + val);
+    if (val != null) {
+      try {
+	return Integer.parseInt(val);
+      } catch (Exception e) {
+	System.out.println("Can not parse value: " + val);
+      }
+    }
+    return defVal;
+  }
+
+  private void loadGamesList() {
+    System.out.println("Trying to load games list");
+    try {
+      URL url = getResource("games.txt");
+      LineNumberReader reader =
+	new LineNumberReader(new InputStreamReader(url.openConnection().getInputStream()));
+
+      // game loop...
+      String games;
+      String disk;
+      files = new Vector();
+      while ((disk = reader.readLine()) != null) {
+	disk = disk.trim();
+	if (disk.toLowerCase().endsWith(".prg") ||
+	    disk.toLowerCase().endsWith(".p00")) {
+//        System.out.println("Adding PGM file: " + disk);
+	  files.addElement(disk);
+	  files.addElement(disk);
+	} else {
+//        System.out.println("reading games from disk: " + disk);
+	  games = reader.readLine();
+	  if (games != null) {
+	    games = games.trim();
+	    StringTokenizer stok = new StringTokenizer(games, ",");
+	    while (stok.hasMoreElements()) {
+	      String game = stok.nextToken();
+	      files.addElement(disk);
+	      files.addElement(game);
+	      System.out.println("Adding: " + game);
+	    }
+	  }
+	}
+      }
+    } catch (Exception e) {
+      System.out.println("Can not load games..." + e);
+      e.printStackTrace();
+      System.out.println( "No games to load...");
+    }
+  }
+
+  // -------------------------------------------------------------------
+  // For scripting!
+  // -------------------------------------------------------------------
+
+  public void setColorSet(int i) {
+    i = i % VICConstants.COLOR_SETS.length;
+    screen.setColorSet(i);
+  }
+
+  public void poke(int address, int data) {
+    if (address < 0xd000 || address >= 0xe000) {
+      int[] memory = cpu.getMemory();
+      memory[address & 0xffff] = data & 0xff;
+    } else {
+      cpu.poke(address & 0xffff, data & 0xff);
+    }
+  }
+
+  public int peek(int address) {
+    int[] memory = cpu.getMemory();
+    return memory[address & 0xffff];
+  }
+
+  public void pause() {
+    cpu.setPause(true);
+  }
+
+  public void unpause() {
+    cpu.setPause(false);
+  }
+
+  private void loadProgram(int item) {
+    String disk = (String) files.elementAt(item * 2);
+    String name = (String) files.elementAt(item * 2 + 1);
+
+    System.out.println("Index:" + item + " -> " + disk + " " + name);
+    loadProgram(disk, name);
+  }
+
+  private boolean loadProgram(String disk, String name) {
+    boolean em1541 = require1541;
+    if (disk.startsWith("@")) {
+      disk = disk.substring(1);
+      em1541 = true;
+    }
+    if (disk != currentDisk) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.